ebml/EbmlElement.h

```diff
@@ -222,18 +228,58 @@ class EBML_DLL_API EbmlCallbacks {
 */
 class EBML_DLL_API EbmlSemantic {
   public:
+    struct DefaultValues;
+
     constexpr EbmlSemantic(bool aMandatory, bool aUnique, const EbmlCallbacks & aCallbacks)
-      :Mandatory(aMandatory), Unique(aUnique), Callbacks(aCallbacks) {}
+      :Mandatory(aMandatory), Unique(aUnique), defaultValue(), Callbacks(aCallbacks) {}
+
+    constexpr EbmlSemantic(bool aMandatory, bool aUnique, const DefaultValues &def, const EbmlCallbacks & aGetCallbacks)
+      :Mandatory(aMandatory), Unique(aUnique), defaultValue(def), Callbacks(aGetCallbacks) {}
 
         inline bool IsMandatory() const { return Mandatory; }
         inline bool IsUnique() const { return Unique; }
         inline EbmlElement & Create() const { return EBML_INFO_CREATE(Callbacks); }
         inline explicit operator const EbmlCallbacks &() const { return Callbacks; }
         inline EbmlCallbacks const &GetCallbacks() const { return Callbacks; }
 
+    enum DefaultType {
+      NO_DEFAULT,
+      UINTEGER,
+      SINTEGER,
+      DATE,
+      DOUBLE,
+      STRING,
+      UNISTRING,
+    };
+   struct DefaultValues {
+      DefaultType type;
+      union {
+        std::uint64_t  u64;
+        std::int64_t   i64;
+        double         f;
+        const char*    s;
+        const wchar_t* ws;
+      };
+
+      constexpr DefaultValues(void) : u64(0), type(DefaultType::NO_DEFAULT) {}
+      constexpr DefaultValues(const DefaultValues &) = default;
+      constexpr DefaultValues(std::uint64_t u) : u64(u), type(DefaultType::UINTEGER) {}
+      constexpr DefaultValues(std::int64_t u, bool date = false) : i64(u), type(date ? DefaultType::UINTEGER : DefaultType::DATE) {}
+      constexpr DefaultValues(double d) : f(d), type(DefaultType::DOUBLE) {}
+      constexpr DefaultValues(const char *d) : s(d), type(DefaultType::STRING) {}
+      constexpr DefaultValues(const wchar_t *d) : ws(d), type(DefaultType::UNISTRING) {}
+
+      bool HasDefault() const { return type != DefaultType::NO_DEFAULT; }
+    };
+
+    inline const DefaultValues & DefaultValue() const { return defaultValue; }
+
+#if defined(EBML_STRICT_API)
```

Comment:
`EBML_STRICT_API` has been removed with one of the commits over the last couple of days. Just always use `private:`

Reply:
Indeed, good catch.